src/command/dev/do_dev.rs

```diff
@@ -146,7 +145,7 @@ impl Dev {
 
             // start the interprocess socket health check in the background
             let health_messenger = follower_messenger.clone();
-            tp.spawn(move || {
+            tokio::task::spawn_blocking(move || {
                 let _ = health_messenger.health_check().map_err(|_| {
                     eprintln!("{}shutting down...", Emoji::Stop);
                     std::process::exit(1);
```

Comment:
Exiting handles still-running tasks differently than still-running threads. I don't remember the details but we'll want to test (if we haven't already) that expected things actually shut down. Notably, that the router subprocess doesn't keep on running.

Reply:
I think we're good!

```
 aaronarinder@Aarons-MBP  ~/apollo/rover   asyncification  cargo rover dev --graph-ref betelgeuse-
countries@main --profile staging-test
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.27s
     Running `target/debug/rover dev --graph-ref 'betelgeuse-countries@main' --profile staging-test`
⌛  retrieving subgraphs remotely from betelgeuse-countries@main
✅  supergraph config loaded successfully
⚠️  Do not run this command in production! ⚠️  It is intended for local development.
🛫 starting a session with the 'countries' subgraph
🛫 starting a session with the 'name' subgraph
🛫 starting a session with the 'private-countries' subgraph
🎶 composing supergraph with Federation v2.8.2
WARN: telemetry.instrumentation.spans.mode is currently set to 'deprecated', either explicitly or via 
defaulting. Set telemetry.instrumentation.spans.mode explicitly in your router.yaml to 'spec_compliant
' for log and span attributes that follow OpenTelemetry semantic conventions. This option will be defa
ulted to 'spec_compliant' in a future release and eventually removed altogether
🚀 your supergraph is running! head to http://localhost:4000 to query your supergraph
✅  successfully composed after adding the 'private-countries' subgraph
^C
✋ shutting down the `rover dev` session and all attached processes...
 ✘ aaronarinder@Aarons-MBP  ~/apollo/rover   asyncification  ps aux | grep "router"
aaronarinder     58299   0.0  0.0        0      0 s006  ?+    1:39PM   0:00.00 grep --color=auto --exc
lude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=
.idea --exclude-dir=.tox router
```